CONTRIBUTING.rst

```diff
@@ -56,12 +56,25 @@ These results will list any failed tests. Coverage reports and any screenshots
 from failed web client tests will be attached to the build as artifact files. You can reach your
 build by clicking the build status link on your GitHub PR.
 
+Tracking Public Symbols
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Public symbol addition and removal is tracked in Girder through the ``scripts/publicNames.txt`` file
```

Comment:
I would start with

"Adding new public symbols to Girder's python library should only be done intentionally, as doing so increases the surface of the API and introduces a maintenance burden. Public symbols are packages, modules, and symbols within those modules that do not start with an underscore character. To help with this goal, public symbol addition and removal is tracked automatically as part of our CI process, with the full list of symbols residing in ``scripts/publicNames.txt``."